src/ostree/ot-remote-builtin-modify.c

```diff
@@ -0,0 +1,149 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright © 2017 Endless Mobile, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ * Authors:
+ *  - Dan Nicholson <nicholson@endlessm.com>
+ */
+
+#include "config.h"
+
+#include "otutil.h"
+#include "ot-tool-util.h"
+
+#include "ot-main.h"
+#include "ot-remote-builtins.h"
+
+static char *opt_url;
+static char **opt_set;
+static gboolean opt_no_gpg_verify;
+static char *opt_contenturl;
+#ifdef OSTREE_ENABLE_EXPERIMENTAL_API
+static char *opt_collection_id;
+#endif  /* OSTREE_ENABLE_EXPERIMENTAL_API */
+static char *opt_sysroot;
+static char *opt_repo;
+
+/* ATTENTION:
+ * Please remember to update the bash-completion script (bash/ostree) and
+ * man page (man/ostree-remote.xml) when changing the option list.
+ */
+
+static GOptionEntry option_entries[] = {
+  { "url", 0, 0, G_OPTION_ARG_STRING, &opt_url, "Set remote URL", "URL" },
+  { "set", 0, 0, G_OPTION_ARG_STRING_ARRAY, &opt_set, "Set config option KEY=VALUE for remote", "KEY=VALUE" },
+  { "no-gpg-verify", 0, 0, G_OPTION_ARG_NONE, &opt_no_gpg_verify, "Disable GPG verification", NULL },
+  { "contenturl", 0, 0, G_OPTION_ARG_STRING, &opt_contenturl, "Use URL when fetching content", "URL" },
+#ifdef OSTREE_ENABLE_EXPERIMENTAL_API
+  { "collection-id", 0, 0, G_OPTION_ARG_STRING, &opt_collection_id,
+    "Globally unique ID for this repository as an collection of refs for redistribution to other repositories", "COLLECTION-ID" },
+#endif  /* OSTREE_ENABLE_EXPERIMENTAL_API */
+  { "repo", 0, 0, G_OPTION_ARG_FILENAME, &opt_repo, "Path to OSTree repository (defaults to /sysroot/ostree/repo)", "PATH" },
+  { "sysroot", 0, 0, G_OPTION_ARG_FILENAME, &opt_sysroot, "Use sysroot at PATH (overrides --repo)", "PATH" },
+  { NULL }
+};
+
+gboolean
+ot_remote_builtin_modify (int argc, char **argv, GCancellable *cancellable, GError **error)
+{
+  g_autoptr(GOptionContext) context = NULL;
+  g_autoptr(OstreeSysroot) sysroot = NULL;
+  g_autoptr(OstreeRepo) repo = NULL;
+  const char *remote_name;
+  char **iter;
+  g_autoptr(GVariantBuilder) optbuilder = NULL;
+  g_autoptr(GVariant) options = NULL;
```

Comment:
This is minor, though note that we tend to declare things as they are assigned now. For iteration variables like `iter` and `i`, you can just declare in the for loop setup.

Reply:
I thought about that. I just copied remote-builtin-add, but I can make these more modern-y.